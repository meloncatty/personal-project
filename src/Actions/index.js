import {apiKey} from '../apiKey.js'

export const resultsSuccess = (results) => ({
  type: 'QUERY_RESULTS_SUCCESS',
  results
})

export const resultsAreLoading = (bool) => ({
  type: 'RESULTS_ARE_LOADING',
  resultsLoading: bool
})

export const resultsHaveErrored = (bool) => ({
  type: 'RESULTS_HAVE_ERRORED',
  resultsErrored: bool
})

export const fullArticleLoading = (bool) => ({
  type: 'FULL_ARTICLE_LOADING',
  articleLoading: bool
})

export const fullArticleErrored = (bool) => ({
  type: 'FULL_ARTICLE_ERRORED',
  articleErrored: bool
})

export const fullArticleSuccess = (result) => ({
  type: 'FULL_ARTICLE_SUCCESS',
  result
})

export const userAuthentication = (userData) => ({
  type: 'USER_AUTHENTICATION',
  userAuth: userData
})

export const isUserSignedIn = (bool) => ({
  type: 'IS_USER_SIGNED_IN',
  isUserSignedIn: bool
})

export const captureUserArticles = (articles) => ({
  type: 'CAPTURE_USER_ARTICLES',
  userArticles: articles
})

export const fetchArticles = (query) => {
  return async (dispatch) => {
    try {
      dispatch(resultsSuccess([]))
      dispatch(resultsHaveErrored(false))
      dispatch(resultsAreLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/search/${query}?page=1&pageSize=10&metadata=true&fulltext=false&citations=true&similar=false&duplicate=false&urls=true&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(resultsAreLoading(false))
      const articles = await response.json()
      dispatch(resultsSuccess(articles.data))
    } catch (error) {
      dispatch(resultsHaveErrored(true))
      dispatch(resultsAreLoading(false))
    }
  }
}

export const fetchFullText = (id) => {
  return async (dispatch) => {
    try {
      dispatch(fullArticleLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/get/${id}?metadata=true&fulltext=true&citations=false&similar=false&duplicate=false&urls=false&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(fullArticleLoading(false))
      const article = await response.json()
      dispatch(fullArticleSuccess(article.data))
    } catch (error) {
      dispatch(fullArticleErrored(true))
      dispatch(fullArticleLoading(false))
    }
  }
}
