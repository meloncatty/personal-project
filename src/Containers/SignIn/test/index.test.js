import React from 'react'
import { SignIn, mapDispatchToProps } from '../'
import { shallow } from 'enzyme'
import {userAuthentication, isUserSignedIn} from '../../../Actions'

describe('SignIn', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SignIn />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have default state', () => {
    const wrapper = shallow(<SignIn />)
    const expected = {
      email: '',
      password: '',
      error: null
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('should not submit form if fields are invalid', () => {
    const wrapper = shallow(<SignIn />)
      const mockEvent = {target: {name: 'password', value: null}}

      wrapper.find('.password-sign-in').simulate('change', mockEvent)

      expect(wrapper.find('button').is('[disabled]')).toBe(true)
  })

  describe('handleSignInChange', () => {
    it('should handle password input change', () => {
      const wrapper = shallow(<SignIn />)
      const mockEvent = {target: {name: 'password', value: 'password'}}

      wrapper.find('.password-sign-in').simulate('change', mockEvent)

      expect(wrapper.state('password')).toEqual(mockEvent.target.value)
    })

    it('should handle email input change', () => {
      const wrapper = shallow(<SignIn />)
      const mockEvent = {target: {name: 'email', value: 'email'}}

      wrapper.find('.email').simulate('change', mockEvent)

      expect(wrapper.state('email')).toEqual(mockEvent.target.value)
    })
  })

  describe('handleSignInSubmit', () => {
    it('should call userAuthentication', () => {
      const wrapper = shallow(<SignIn />)
      wrapper.instance().handleSignInSubmit

      expect(wrapper.instance().userAuthentication).toHaveBeenCalled
    })

    it('should call isUserSignedIn', () => {
      const wrapper = shallow(<SignIn />)
      wrapper.instance().handleSignInSubmit

      expect(wrapper.instance().isUserSignedIn).toHaveBeenCalled
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with user UID when handleSignInSubmit is called', () => {
      const wrapper = shallow(<SignIn />)
      const mockDispatch = jest.fn()
      const actionToDispatch = userAuthentication('userUID')
      const mockEvent = {preventDefault: () => {}}
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.userAuthentication('userUID')

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call dispatch with true is user is signed in', () => {
      const wrapper = shallow(<SignIn />)
      const mockDispatch = jest.fn()
      const actionToDispatch = isUserSignedIn(true)
      const mockEvent = {preventDefault: () => {}}
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.isUserSignedIn(true)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
})