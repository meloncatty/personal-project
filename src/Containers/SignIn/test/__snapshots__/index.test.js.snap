// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="sign-in-form"
          onSubmit={[Function]}
>
          <label>
                    Sign In
          </label>
          <input
                    className="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
          />
          <input
                    className="password-sign-in"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign In
          </button>
</form>,
        <Link
          replace={false}
          to="/passwordForget"
>
          Forgot Password?
</Link>,
      ],
      "className": "form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Sign In
</label>,
            <input
              className="email"
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
/>,
            <input
              className="password-sign-in"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign In
</button>,
            null,
          ],
          "className": "sign-in-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password-sign-in",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
          null,
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Forgot Password?",
          "replace": false,
          "to": "/passwordForget",
        },
        "ref": null,
        "rendered": "Forgot Password?",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="sign-in-form"
            onSubmit={[Function]}
>
            <label>
                        Sign In
            </label>
            <input
                        className="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="text"
            />
            <input
                        className="password-sign-in"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign In
            </button>
</form>,
          <Link
            replace={false}
            to="/passwordForget"
>
            Forgot Password?
</Link>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Sign In
</label>,
              <input
                className="email"
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
/>,
              <input
                className="password-sign-in"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign In
</button>,
              null,
            ],
            "className": "sign-in-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password-sign-in",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            null,
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Forgot Password?",
            "replace": false,
            "to": "/passwordForget",
          },
          "ref": null,
          "rendered": "Forgot Password?",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
